### PR TITLE
Componentize the settings page

### PR DESCRIPTION
**Pull request by TheColaber**
**Changes**

As of writing this, this PR converts the addon body and the addon settings into components.

**Reason for changes**

For organizations so functions for just settings or just addons stay in one place rather than with everything else. Also, so we don't have to pass in `addon` or `setting` into functions.

**Tests**

Probably needs more testing just in case.

**Discuss**
What else might need to be componentized?

### CODE REVIEW

**Comment by GrahamSH-LLK** [comment]
This is awesome! One thing I want to consider is figuring out how to have separate files for components.

**Comment by GrahamSH-LLK** [comment]
> What else might need to be componentized?

Categories and icons

**Comment by WorldLanguages** [comment]
> One thing I want to consider is figuring out how to have separate files for components.

Keep in mind that a file with an inline `<script></script>` tag won't work because of the extension CSP. We'll need separate .html and .js files if we want to have components as separate files (and if we want to have separated `Vue.extend()`s as well

File: webpages/settings/index.js [hunk 1]
```diff
@@ -474,64 +579,28 @@ const vue = (window.vue = new Vue({
...
-          this.updateOption(property.id, property.default, addon);
-        }
-        console.log(`Loaded default values for ${addon.id}`);
-      }
-    },
     closePickers(e, leaveOpen) {
-      for (let child of this.$children) {
-        if (child.isOpen && child.canCloseOutside && e.isTrusted && child.color && child !== leaveOpen) {
-          child.toggle(child.addon, child.setting, false);
+      for (let addon of this.$children) {
```
Comment (WorldLanguages) [inline webpages/settings/index.js:583]: Maybe `this.$children.filter(c => c.$options.name === "addon-body")`

Comment (WorldLanguages) [reply webpages/settings/index.js:583]: This is ok for now :P

**Pull request merged**